crates/storage/provider/src/changesets_utils/trie.rs

```diff
@@ -1,51 +1,42 @@
-use alloy_primitives::B256;
 use itertools::{merge_join_by, EitherOrBoth};
-use reth_db_api::{cursor::DbDupCursorRO, tables, DatabaseError};
-use reth_trie::{BranchNodeCompact, Nibbles, StorageTrieEntry, StoredNibblesSubKey};
+use reth_db_api::DatabaseError;
+use reth_trie::{trie_cursor::TrieCursor, BranchNodeCompact, Nibbles};
 use std::cmp::{Ord, Ordering};
 
-/// Combines a sorted iterator of trie node paths and a [`tables::StoragesTrie`] cursor into a new
+/// Combines a sorted iterator of trie node paths and a storage trie cursor into a new
 /// iterator which produces the current values of all given paths in the same order.
 #[derive(Debug)]
 pub struct StorageTrieCurrentValuesIter<'cursor, P, C> {
-    /// The address of the account whose storage is being iterated.
-    hashed_address: B256,
     /// Sorted iterator of node paths which we want the values of.
     paths: P,
-    /// Cursor over [`tables::StoragesTrie`].
+    /// Storage trie cursor.
     cursor: &'cursor mut C,
     /// Current value at the cursor, allows us to treat the cursor as a peekable iterator.
     cursor_current: Option<(Nibbles, BranchNodeCompact)>,
 }
 
 impl<'cursor, P, C> StorageTrieCurrentValuesIter<'cursor, P, C>
 where
-    C: DbDupCursorRO<tables::StoragesTrie>,
+    P: Iterator<Item = Nibbles>,
+    C: TrieCursor,
 {
     /// Instantiate a [`StorageTrieCurrentValuesIter`] from a sorted paths iterator and a cursor.
-    pub fn new(
-        hashed_address: B256,
-        paths: P,
-        cursor: &'cursor mut C,
-    ) -> Result<Self, DatabaseError> {
-        let mut new_self = Self { hashed_address, paths, cursor, cursor_current: None };
+    pub fn new(paths: P, cursor: &'cursor mut C) -> Result<Self, DatabaseError> {
+        let mut new_self = Self { paths, cursor, cursor_current: None };
         new_self.seek_cursor(Nibbles::default())?;
         Ok(new_self)
     }
 
     fn seek_cursor(&mut self, path: Nibbles) -> Result<(), DatabaseError> {
-        self.cursor_current = self
-            .cursor
-            .seek_by_key_subkey(self.hashed_address, StoredNibblesSubKey(path))?
-            .map(|e| (e.nibbles.0, e.node));
+        self.cursor_current = self.cursor.seek(path)?;
         Ok(())
     }
 }
```

Comment:
do you think it makes sense to bring the abstraction up? since its now a wrapper